x/auth/ante/unorderedtx/manager.go

```diff
@@ -23,15 +37,31 @@ type Manager struct {
 	// doneCh allows us to ensure the purgeLoop has gracefully terminated prior to closing
 	doneCh chan struct{}
 
+	// dataDir defines the directory to store unexpired unordered transactions
+	//
+	// XXX: Note, ideally we avoid the need to store unexpired unordered transactions
+	// directly to file. However, store v1 does not allow such a primitive. But,
+	// once store v2 is fully integrated, we can remove manual file handling and
+	// store the unexpired unordered transactions directly to SS.
+	//
+	// Ref: https://github.com/cosmos/cosmos-sdk/issues/18467
+	dataDir string
+
 	mu sync.RWMutex
 	// txHashes defines a map from tx hash -> TTL value, which is used for duplicate
 	// checking and replay protection, as well as purging the map when the TTL is
 	// expired.
 	txHashes map[TxHash]uint64
 }
 
-func NewManager() *Manager {
+func NewManager(dataDir string) *Manager {
+	path := filepath.Join(dataDir, dirName)
```

Comment:
whats the difference with putting the file creation from app.go here vs in app.go?

Reply:
The idea, like WASMVM, is to have the "component" be responsible or "own" the directory/location of where data is written. Also, less work for the app to do.

Reply:
does that mean this https://github.com/cosmos/cosmos-sdk/pull/18739/files#r1430642896 would get moved into here?

Reply:
Sorry, which file?